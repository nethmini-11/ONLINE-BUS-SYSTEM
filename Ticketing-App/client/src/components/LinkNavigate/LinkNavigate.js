import { useState } from "react";
import PropTypes from "prop-types";
import React from 'react';
const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
};

const LinkNavigate = ({ page, children }) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={page || "#"}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

LinkNavigate.defaultProps = {
  page: "localhost:3001/managetimetable",

};

LinkNavigate.propTypes = {
  page: PropTypes.string,
};

export default LinkNavigate;


