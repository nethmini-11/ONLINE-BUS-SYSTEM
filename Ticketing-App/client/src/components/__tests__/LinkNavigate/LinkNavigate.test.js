import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import LinkNavigate from '../../LinkNavigate/LinkNavigate'
import React from 'react';
afterEach(() => {
  cleanup();
});

test("renders correctly", () => {
  const component = renderer.create(
    <LinkNavigate page="localhost:3001/managetimetable">Link Navigate</LinkNavigate>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


