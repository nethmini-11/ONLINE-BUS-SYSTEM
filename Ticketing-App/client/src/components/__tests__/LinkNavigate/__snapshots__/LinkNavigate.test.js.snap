// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="localhost:3001/managetimetable"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Made with ❤️ - Aseem
</a>
`;
